File: FE/AL-homepage/src/types/subject.ts
export interface Subject {
  subject_name: string
  subject_nick: string
  subject_id: number
}

export interface CreateSubjectRequest {
  subject_name: string
  subject_nick: string
}

export interface UpdateSubjectRequest {
  subject_name?: string
  subject_nick?: string
}

export interface GetAllSubjectsParams {
  skip?: number
  limit?: number
}

File: BE/avalon/app/models/subject.py
from sqlalchemy import Column, Integer, String
from .base import Base

class Subject(Base):
    __tablename__ = "subjects"
    subject_id = Column(Integer, primary_key=True, autoincrement=True)
    subject_name = Column(String(100), nullable=False)
    subject_nick = Column(String(20), nullable=False)